ska_builder.py

```diff
@@ -203,9 +213,49 @@ def build_list_packages(pkg_names, args, src_dir, build_dir):
         raise ValueError("Packages {} failed".format(",".join(failures)))
 
 
+def overwrite_skare3_version(current_version, new_version, pkg_path):
+    meta_file = pkg_path / 'meta.yaml'
+    with open(meta_file) as fh:
+        t = jinja2.Template(fh.read())
+    text = (t.render(SKA_PKG_VERSION='$SKA_PKG_VERSION',
+                     SKA_TOP_SRC_DIR='$SKA_TOP_SRC_DIR'))
+    if version.parse(yaml.__version__) < version.parse("5.1"):
+        data = yaml.load(text)
+    else:
+        data = yaml.load(text, Loader=yaml.BaseLoader)
+    if str(data['package']['version']) == str(current_version):
+        data['package']['version'] = new_version
+        for i in range(len(data['requirements'])):
```

Comment:
Changing the version in the run requirements is confusing me. 

For example, consider building `ska3-flight`. In order to get into this function we need that `current_version` matches the package version in the `ska3-flight/meta.yaml`. But then this code looks for any other run requirement package that starts with `ska3-` and if that version also matches `current_version` then it gets replaced.  So if `ska3-core` happens to have the same version as `ska3-flight` then it gets replaced?

Reply:
Yes, because the intent of this option is to replace a version by another in a skare3 release.

All ska3-* packages with matching version would have their version changed, and you would want their inter-dependencies also changed. If you are replacing the version of ska3-flight, and ska3-core appears as a dependency with the same version, then presumably you also need to build ska3-core with the replaced version.

In other words. Suppose you have ska3-core==2020.13 and ska3-flight 2020.13. Then ska3-flight will depend on ska3-core==2020.13. The way it is used is that all appearances of 2020.13 are replaced (by something like 2020.13rc1). Do you see any case where one would replace ska3-flight version and not ska3-core?

Reply:
But does this mean that version numbers of ska3-core and ska3-flight need to track each other in order for this scheme to work? I may still just be generally confused about the whole workflow.

Reply:
No. You can release ska3-flight 2020.13rc1 which depends on ska3-core 2020.12. This option would replace appearances of 2020.13 by 2020.13rc1, and ska3-core does not match, so it is unchanged.

If you were to update both of them, both ska3-flight/core are 2020.13rc1, then ska3-flight must depend on ska3-core 2020.13, so the ska3-core version in the requirements is also replaced by 2020.13rc1.